frappe/email/receive.py

```diff
@@ -306,12 +307,18 @@ def set_content_and_type(self):
 		else:
 			self.content, self.content_type = EmailReplyParser.parse_reply(self.text_content), 'text/plain'
 
-	def process_part(self, part):
+	def process_part(self, part,last):
 		"""Parse email `part` and set it to `text_content`, `html_content` or `attachments`."""
 		content_type = part.get_content_type()
 		charset = part.get_content_charset()
 		if not charset: charset = self.get_charset(part)
 
+		if content_type == 'message/rfc822' or last[0] == True: #test to see if within a attached message
+			if self.get_attached_message(part): #check and get for email headers
```

Comment:
unfortunately that works great for case 1 but the case 2 has another layer in the mime format before the header